pkgs/desktops/gnome-3/core/mutter/default.nix

```diff
@@ -84,6 +87,7 @@ let self = stdenv.mkDerivation rec {
   nativeBuildInputs = [
     desktop-file-utils
     gettext
+    mesa # needed for gbm
```

Comment:
Looks like gbm might be separated out of mesa in the future https://github.com/NixOS/nixpkgs/pull/118479

Reply:
i don't even know what that is. i just applied the fix that was used in other packages :D

having a separate package sounds resonable